### PR TITLE
Setting up a development environment for all CinemaNet models + Docs

### PR DESCRIPTION
**Pull request by rsomani95**
**This PR is a selective port from the [vision_models](https://github.com/Synopsis/CinemaNet/tree/vision_models) branch, moving materials that will be useful in training of all CinemaNet models going forward.**

---

Using `nbdev` to use Jupyter notebooks to create a package that will make the training of new models easier + provide tools for faster fine-tuning. Conveniently, we can also generate documentation from these notebooks using `nbdev`. Pushing the package to `pypi` can also be done with minimal additional effort. 
[Nbdev Website](http://nbdev.fast.ai/).

The package details are in the `settings.ini` file.

The `nbs/` folder contains all the notebooks and the `docs/` folder the documentation. We can deploy a GitHub Pages with minimal effort whenever we want to. 

---

Brief description of what the library/notebooks contain:
* `nbs/utils_01_fastai_custom_head` -- By default, `fastai` adds a customised head for any pretrained model. Here, one line of code is changed to make these custom heads CoreML compatible.
* `nbs/utils_02_wandb_callback` -- A custom `wandb` callback that adds functionality to log confusion matrices at the end of every training loop.
* `nbs/utils_03_pytorch_to_coreml` -- Makes porting PyTorch to CoreML possible with one high level function call. Lower level functions can be used to convert to ONNX first, and then to CoreML. 
* `nbs/utils_04_mobilenet_v2_basic` -- A basic training loop + some convenience functions for loading data, setting up metrics, callbacks, data augmentations, etc.

### CODE REVIEW

**Comment by vade** [comment]
Hey hey - I noticed this has some Jekyll files and a lot of CSS ? Is this for Jupiter notebooks or documentation? Just curious I saw references to GitHub.io etc and wanted to ensure I wasn't misunderstanding the intent here!

File: CinemaNet/convert.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+import os
+
+from onnx_coreml import convert
+from onnx import onnx_pb
+
+# Cell
+args = dict(is_bgr=False,
+            red_bias   = -(0.485 * 255.0),
+            green_bias = -(0.456 * 255.0),
+            blue_bias  = -(0.406 * 255.0))
```
Comment (vade) [inline CinemaNet/convert.py]: Are these biases taken from a particular architecture / training method ? These dont match other conversion constants ive seen for CoreML/Mobilenet?

File: CinemaNet/convert.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    scale_layer = nn_spec.layers.add()
+    scale_layer.name = 'scaler'
+    scale_layer.input.append('input.1')
+    scale_layer.output.append('input.1_scaled')
+
+    params = scale_layer.scale
+    scale_R = 1.0 / (0.229 * 255.0)
+    scale_G = 1.0 / (0.224 * 255.0)
+    scale_B = 1.0 / (0.225 * 255.0)
+    params.scale.floatValue.extend([scale_R, scale_G, scale_B])
```
Comment (vade) [inline CinemaNet/convert.py]: Same with comment above

**Comment by rsomani95** [comment]
> Hey hey - I noticed this has some Jekyll files and a lot of CSS ? Is this for Jupiter notebooks or documentation? Just curious I saw references to GitHub.io etc and wanted to ensure I wasn't misunderstanding the intent here!

It's all for the documentation. By default it appends the documentation link to synopsis.github.io/synopsis.video, so the link would look something like synopsis.github.io/cinemanet. We'll tweak this before deploying though.

File: CinemaNet/convert.py (hunk 1 above)
Comment (rsomani95) [reply CinemaNet/convert.py]: They're taken from the MobileNet (or any other architecture) pretrained on ImageNet. These can be seen in the `ONNX ⟶ CoreML (Raw)` section in the `nbs/utils_03_pytorch_to_coreml.ipynb` (not exported to the `convert.py` script.

File: CinemaNet/convert.py (hunk 2 above)
Comment (rsomani95) [reply CinemaNet/convert.py]: ^ Same answer as above

**Pull request merged**